### PR TITLE
Attempt to create inline Commands

### PR DESCRIPTION
**Pull request by omentic**
I'm doing a great deal of things wrong.

### CODE REVIEW

**Comment by omentic** [comment]
Major problems (aside from the methods I'm referencing not existing):
- `configureJoystickBindings` and `configureButtonBoardBindings` can't find the Joysticks or Subsystems
- `void` is passed to a bunch of the FunctionalCommands - my intent is to pass an empty method...
- Even if I throw the subsystems in the `configure` methods and replace `void` with some method, all my inline commands are undefined...

**Comment by dbadb** [comment]
As we discussed at length last night, there are pros and cons with this approach:

pro:
1. it's concise
2. it removes the need for lots of external command files for trivial applications.

con:
1. it's a little hard on the eyes - there isn't a lot of room for comments in this lambda function style.
2. the consolidation of commands into a single file has the down side of:
 2a. reduced "encapsulation"
 2b. increased likelihood of git conflicts.

We discussed goals for code factoring.

1. we'd like a few files, each with clear ownership (perhaps one per subsystem)
2. we need to make sure that programmers understand/embrace the separation of command from subsystem.
3. we really need programmers to understand the execution model of commands and command-groups.   This overlaps with questions of "default commands", diagnostic requirements, etc.

**Comment by dbadb** [comment]
btw: i'll be attempting to produce an example command factory today.  I'll let you know how it goes.

**Comment by pietroglyph** [comment]
I’ll be excited to see what you produce. Having multiple examples that we can compare should help us come to a conclusion on this issue (especially because I’m not really sure what the right way is going to be.)

**Comment by dbadb** [comment]
No rocket science here..

https://docs.oracle.com/javase/tutorial/java/javaOO/anonymousclasses.html

may be a useful tool for defining multiple classes in a single file.

another might be simply nested classes.

**Comment by dbadb** [comment]
Here's an example:

https://github.com/Spartronics4915/2020-InfiniteRecharge/blob/master/src/main/java/com/spartronics4915/frc2020/commands/ExampleCommandFactory.java

**Comment by omentic** [comment]
We've decided to go with a single individual file per-subsystem for all Commands to go in. Structurally, it'll be like @dbadb 's `Test3` and `Test4`. If so desired, we can also extend `InstantCommand` or any other convenience feature. 

Austin will have the `ClimberCommands.java` file written early next meeting (Wednesday) as an example for those working on commands.

**Comment by omentic** [comment]
@dbadb How would I implement something like StartEndCommand as a subclass?

```
public class ExampleCommandFactory
{
...
    public class ExampleSimpleCommand extends StartEndCommand
    {
        ExampleSimpleCommand() 
        {
            // This is not correct
            new StartEndCommand(mSubsystem::extend, mSubsystem::retract, mSubsystem); 
        }
    }
}
```

I know that I could define `execute()` and `end()`, but it seems like I should be able to avoid that...

**Comment by dbadb** [comment]
You must manually invoke the parent constructor via super(...)

**Comment by dbadb** [comment]
Here's an example:

https://github.com/Spartronics4915/2020-InfiniteRecharge/commit/7b9b1132d8289f9734e91f17a62266697cd26f5d

File: src/main/java/com/spartronics4915/frc2020/RobotContainer.java [hunk 1]
```diff
@@ -41,56 +41,88 @@ public AutoMode(String name, Command command)
...
     /**
      * The container for the robot. Contains subsystems, OI devices, and commands.
      */
     public RobotContainer()
     {
+        mClimber = new Climber();
+        mIntake = new Intake();
+        mLauncher = new Launcher();
+        mPanelRotator = new PanelRotator();
+        mClimberCommands = new ClimberCommands(mClimber);
```
Comment (j-james) [inline src/main/java/com/spartronics4915/frc2020/RobotContainer.java]: This code builds, but passing the subsystem to both the containing class and the Command seems redundant.

File: src/main/java/com/spartronics4915/frc2020/RobotContainer.java [hunk 2]
```diff
@@ -41,56 +41,88 @@ public AutoMode(String name, Command command)
...
     /**
      * The container for the robot. Contains subsystems, OI devices, and commands.
      */
     public RobotContainer()
     {
+        mClimber = new Climber();
+        mIntake = new Intake();
+        mLauncher = new Launcher();
+        mPanelRotator = new PanelRotator();
+        mClimberCommands = new ClimberCommands();
```
Comment (omentic) [inline src/main/java/com/spartronics4915/frc2020/RobotContainer.java:76]: This code builds, but passing the subsystem to both the containing class and the Command seems redundant.

Comment (omentic) [reply src/main/java/com/spartronics4915/frc2020/RobotContainer.java:76]: @dbadb Is there any functional difference between passing the subsystem to each Command individually vs. as below?

(set up now with `ClimberCommands` and `PanelRotatorCommands`)

**Pull request merged**